src/Components/Notes/Notes.jsx

```diff
@@ -51,20 +57,44 @@ export default function Notes() {
             username: issue.user.login,
             avatarUrl: issue.user.avatar_url,
             numberOfComments: issue.comments,
+            synchedNote: true,
           })
         })
-        if (fetchedNotes.length > 0) {
-          setNotes(fetchedNotes)
+
+        let synchedCreatedNotes = []
+        if (createdNotes !== null) {
+          synchedCreatedNotes =
+          createdNotes.filter(
+              (createdNote) => !fetchedNotes.some(
+                  (fetchedNote) =>
+                    createdNote.title === fetchedNote.title &&
+                    createdNote.body === fetchedNote.body ),
+          )
+          // update the list of created notes
+          setCreatedNotes(synchedCreatedNotes)
+        }
+
+        const combinedSynchedNotes = [
+          ...synchedCreatedNotes,
+          ...fetchedNotes,
+        ]
+
+        if (deletedNotes !== null) {
+          const filteredDeleted =
+          combinedSynchedNotes.filter(
+              (synchedNote) => !deletedNotes.includes(synchedNote.number),
+          )
+          setNotes(filteredDeleted)
         } else {
-          setNotes([])
+          setNotes(combinedSynchedNotes)
         }
       } catch (e) {
         debug().warn('failed to fetch notes', e)
       }
     }
-
     fetchNotes()
-  }, [setNotes, repository, accessToken])
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you share why this ESLint rule should be disabled here rather than listing the hook's dependencies?

Reply:
included selectedNoteId - missing dependency